internal/controllers/module_nmc_reconciler.go

```diff
@@ -78,13 +81,21 @@ func (mnr *ModuleNMCReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	if mod.GetDeletionTimestamp() != nil {
 		//Module is being deleted
+		if err = mnr.nsHelper.tryRemovingLabel(ctx, mod.Namespace, mod.Name); err != nil {
```

Comment:
Shouldn't we remove the label only after finalizing the module?

Reply:
No, otherwise we may remove the finalizer but fail to remove the label for any reason. In that we will never try to remove the label again.

Reply:
@ybettan we are just trying to prevent namespace deletion, we are not handling the whole SA/RoleBindings etc'. There may a small chance for race condition, but it is better then none